### PR TITLE
Refactoring of parts related to error messages

### PR DESCRIPTION
**Pull request by wafuwafu13**
please see https://github.com/hashicorp/terraform/pull/31725#discussion_r960435200 and https://github.com/hashicorp/terraform/pull/31725#discussion_r960445938

### CODE REVIEW

File: internal/command/init.go [hunk 1]
```diff
@@ -597,6 +597,8 @@ func (c *InitCommand) getProviders(config *configs.Config, state *states.State,
 			c.Ui.Info(fmt.Sprintf("- Installing %s v%s...", provider.ForDisplay(), version))
 		},
 		QueryPackagesFailure: func(provider addrs.Provider, err error) {
+			const summaryFailure = "Failed to query available provider packages"
+			const summaryInvalid = "Invalid provider registry host"
```
Comment (wafuwafu13) [inline internal/command/init.go:601]: other example: https://github.com/hashicorp/terraform/blob/b8bc1dd721e3f80de3e6e1c75f4f964dd39d0b0f/internal/command/init.go#L710

File: internal/providercache/package_install.go [hunk 2]
```diff
@@ -194,13 +194,13 @@ func installFromLocalDir(ctx context.Context, meta getproviders.PackageMeta, tar
 	if suitableHashCount > 0 {
 		if matches, err := meta.MatchesAnyHash(allowedHashes); err != nil {
 			return authResult, fmt.Errorf(
-				"failed to calculate checksum for %s %s package at %s: %s",
-				meta.Provider, meta.Version, meta.Location, err,
+				"failed to calculate checksum for %s v%s package at %s: %s",
+				meta.Provider.ForDisplay(), meta.Version.String(), meta.Location, err,
```
Comment (wafuwafu13) [inline internal/providercache/package_install.go:198]: other example:
https://github.com/hashicorp/terraform/blob/b8bc1dd721e3f80de3e6e1c75f4f964dd39d0b0f/internal/command/init.go#L570

**Comment by wafuwafu13** [comment]
I'll fix https://github.com/hashicorp/terraform/runs/8132260175?check_suite_focus=true

**Comment by crw** [comment]
Hi @wafuwafu13, thanks for this submission! Could you please give us the background on the motivation for these changes? Thanks again!

**Comment by wafuwafu13** [comment]
@crw 
Thank you for reviewing. Updated! https://github.com/hashicorp/terraform/pull/31725#issue-1358545449

**Comment by crw** [comment]
Thanks for this contribution and the follow-on updates. We decided not to merge this PR in review today. We would normally combine a change such as this (that does not have a user impact but improves the code) with a change that is solving a problem for users. The issue is that these changes tend to make future backports more complicated (in terms of merge conflicts), and requires someone from the core dev team manually merging into the older branches.

Thanks for raising this to our attention! The next time we are in this section of code, we will make sure to clean it up per your submission. Thanks again!

**Pull request closed without merging**